src/Plugin/Create/Payload.php

```diff
@@ -0,0 +1,151 @@
+	}
+
+	/**
+	 * @param Request $request
+	 * @return static
+	 */
+	public static function fromRequest(Request $request): static {
+		$self = new static();
+
+		/**
+		 * @var array{
+		 *       CREATE: array{
+		 *           expr_type: string,
+		 *           not-exists: bool,
+		 *           base_expr: string,
+		 *           sub_tree?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string
+		 *           }>
+		 *       },
+		 *       TABLE: array{
+		 *           base_expr: string,
+		 *           name?: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           },
+		 *           create-def?: bool,
+		 *           options?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string,
+		 *               delim: string,
+		 *               sub_tree?: array<array{
+		 *                   expr_type: string,
+		 *                   base_expr: string
+		 *               }>
+		 *           }>
+		 *       },
+		 *       LIKE: array{
+		 *           expr_type: string,
+		 *           table?: string,
+		 *           base_expr: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           }
+		 *       }
+		 *   } $payload
+		 */
+		$payload = Payload::$sqlQueryParser::parse($request->payload);
+
+		$self->destinationTableName = $payload['TABLE']['no_quotes']['parts'][0];
+		$self->sourceTableName = $payload['LIKE']['no_quotes']['parts'][0];
+		return $self;
+	}
+
+	/**
+	 * @param Request $request
+	 * @return bool
+	 */
+	public static function hasMatch(Request $request): bool {
+
+		/**
+		 * @phpstan-var array{
+		 *       CREATE?: array{
+		 *           expr_type: string,
+		 *           not-exists: bool,
+		 *           base_expr: string,
+		 *           sub_tree?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string
+		 *           }>
+		 *       },
+		 *       TABLE?: array{
+		 *           base_expr: string,
+		 *           name?: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           },
+		 *           create-def?: bool,
+		 *           options?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string,
+		 *               delim: string,
+		 *               sub_tree?: array<array{
+		 *                   expr_type: string,
+		 *                   base_expr: string
+		 *               }>
+		 *           }>
+		 *       },
+		 *       LIKE?: array{
+		 *           expr_type: string,
+		 *           table?: string,
+		 *           base_expr: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           }
+		 *       }
+		 *   } $payload
+		 */
+		$payload = Payload::$sqlQueryParser::parse($request->payload);
```

Comment:
We should remove heavy parsing from hasMatch and validate if we should use plugin or not just on simple strchecks as possible (like strpos)